lib/recipe/bloc/recipe_bloc.dart

```diff
@@ -147,7 +147,8 @@ class RecipeBloc extends Bloc<RecipeEvent, RecipeState> {
     List<Recipe> recipes = List.from(state.recipes);
 
     int changedRecipeIndex = recipes.indexOf(event.recipe);
-    List<Ingredient> ingredients = recipes[changedRecipeIndex].ingredients;
+    List<Ingredient> ingredients =
+        List.from(recipes[changedRecipeIndex].ingredients);
```

Comment:
Wow, how come this fixes the bug?

Reply:
Honestly, am not entirely sure but I think generating a new list likely changes an internal id of some sort which leads to the bloc realising state changes as it should - the initial bug being that the state change wasn't being registered despite the list changing.